export/export.go

```diff
@@ -0,0 +1,113 @@
+package export
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+	distrotypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	terra "github.com/terra-money/core/v2/app"
+)
+
+var IsAccountExportWorkerRunning = false
+
+func ExportAllAccounts(app *terra.TerraApp) error {
+	if IsAccountExportWorkerRunning {
+		return fmt.Errorf("exporting is still running")
+	}
+	IsAccountExportWorkerRunning = true
+	go runAccountExportWorker(app)
+	return nil
+}
+
+func ExportCirculatingSupply(app *terra.TerraApp) (sdktypes.Int, error) {
+	height := app.LastBlockHeight()
+	ctx := app.NewContext(true, tmproto.Header{Height: height})
+	time := time.Now()
+	totalVesting := sdktypes.NewInt(0)
+	app.AccountKeeper.IterateAccounts(ctx, func(account authtypes.AccountI) (stop bool) {
+		switch account.(type) {
+		case *vestingtypes.PeriodicVestingAccount:
+			v := account.(*vestingtypes.PeriodicVestingAccount)
+			totalVesting = totalVesting.Add(v.GetVestingCoins(time).AmountOf("uluna"))
+		default:
+			return false
+		}
+		return false
+	})
+	totalSupply, err := app.BankKeeper.TotalSupply(sdktypes.WrapSDKContext(ctx), &banktypes.QueryTotalSupplyRequest{})
+	if err != nil {
+		return sdktypes.Int{}, err
+	}
+	lunaTotalSupply := totalSupply.Supply.AmountOf("uluna")
+	communityPool, err := app.DistrKeeper.CommunityPool(sdktypes.WrapSDKContext(ctx), &distrotypes.QueryCommunityPoolRequest{})
+	if err != nil {
+		return sdktypes.Int{}, err
+	}
+	lunaCommunityPool := communityPool.Pool.AmountOf("uluna").TruncateInt()
+
+	return lunaTotalSupply.Sub(lunaCommunityPool).Sub(totalVesting), nil
+}
+
+func runAccountExportWorker(app *terra.TerraApp) {
+	app.Logger().Info("[export] exporting accounts")
+	height := app.LastBlockHeight()
+	ctx := app.NewContext(true, tmproto.Header{Height: height})
+	// Should use lastest block time
+	time := time.Now()
+	var accounts []string
+	count := 0
+	app.AccountKeeper.IterateAccounts(ctx, func(account authtypes.AccountI) (stop bool) {
+		balance := app.BankKeeper.GetBalance(ctx, account.GetAddress(), "uluna").Amount
+		delegationRewards, err := app.DistrKeeper.DelegationTotalRewards(sdktypes.WrapSDKContext(ctx), &types.QueryDelegationTotalRewardsRequest{
+			DelegatorAddress: account.GetAddress().String(),
+		})
+		if err != nil {
+			panic(err)
+		} else {
+			balance = balance.Add(delegationRewards.Total.AmountOf("uluna").TruncateInt())
+		}
+		switch account.(type) {
+		case *vestingtypes.PeriodicVestingAccount:
```

Comment:
Can we skip other Vesting Account Types?

Reply:
hmm. i think we should include them. let me add them.